setup.py

```diff
@@ -1,45 +1,72 @@
-                     # icc
-                     #'-march=core-avx2',  # for rome on icc
-                     #'-fp-model', 'strict',
-
-                     # clang
-                     #'-stdlib=libc++',
-                     #'-fveclib=SVML',
-                     #'-lsvml'
-
-                     # aocc
-                     #'-std=c++11',
-                     #'-ffast-math',
-                     #'-lamdlibm',
-                     #'-lm',
-                     ]
-
-ext_modules = [
-    Pybind11Extension(
-        "nufft_ls",
-        sorted(glob("src/*.cpp")),  # Sort source files for reproducibility
-        include_dirs=['include/'],
-        extra_compile_args=compile_link_args,
-        extra_link_args=compile_link_args,
-        language='c++',
-    ),
-]
-
-setup(name="nufft_ls",
-    version="0.0.1",
+
+from pybind11.setup_helpers import Pybind11Extension, build_ext, MACOS, WIN, has_flag, intree_extensions
+
+
+ext_modules = intree_extensions(["src/nufft_ls/cpu.cpp"])
+ext_modules[0].include_dirs = ['include/']
+ext_modules[0].language = 'c++'
+
+class custom_build_ext(build_ext):
+    def build_extensions(self):
+        # Handle platform specific optimization flags
+        if WIN:
+            # I don't know what optimization flags would be right on Windows
+            pass
+        else:
+            cflags = []
+            ldflags = []
+            for flag in [
+                "-march=native",
+                "-Ofast",
+                "-fopenmp",
+                "-funroll-loops",
+                "-ffast-math",
+            ]:
+                if has_flag(self.compiler, flag):
+                    cflags.append(flag)
+
+            # None of the rest of these will work on Mac
+            if not MACOS:
+                cc = " ".join(self.compiler.compiler_so[:2])
+                
+                # TODO: this can give a huge speedup when using the Intel compiler on an AMD CPU
+                # but if we're on an Intel CPU, we want -march=native instead. How to detect this?
+                #if "icc" in cc:
+                #    if has_flag(self.compiler, "-march=core-avx2"):
+                #        cflags += ["-march=core-avx2"]
+                
+                if "clang" in cc:
+                    cflags += ["-stdlib=libc++"]
+                    ldflags += ["-stdlib=libc++"]
+                    # These are optimistic/untested commands that might work if one has
+                    # an SVML lib available for Clang to link against
+                    #cflags += ["-fveclib=SVML"]
+                    #ldflags += ["-lsvml"]
+
+                # TODO: AOCC on Clang 13 seems to only support C++11
+                # but the Pybind11 auto-detection doesn't seem to notice
+                #if "aocc" in cc:  # TODO: the AOCC executable is actually just named `clang`...
+                #    cflags += ["-std=c++11"]
+                #    ldflags += ["-lamdlibm", "-lm"]  # not sure if these help...
```

Comment:
AOCC didn't work any magic performance-wise, so I think it's okay to leave this broken.